mlir/docs/Dialects/Shard.md

```diff
@@ -1,28 +1,30 @@
-# 'mesh' Dialect
+# 'shard' Dialect
 
-The `mesh` dialect contains a set of attributes, operations and interfaces that
-are useful for representing sharding and communication on a device mesh
-cluster.
+This dialect contains a set of attributes, operations and interfaces that
+are useful for representing sharding and communication on a device grid.
```

Comment:
Nit: "sharding" of what?
Can we link to some definition or introduction to the concept?